### ISSUE TITLE
Definition of origin: indices_to_points and points_to_indices for voxel

### ISSUE THREAD

**Issue by wkentaro** [mikedh/trimesh#370]
If I run below code

```python
import numpy as np
import trimesh
import trimesh.transformations as tf


matrix = np.ones((2,)*3, dtype=bool)
centers = trimesh.voxel.matrix_to_points(matrix, pitch=0.5, origin=(0,)*3)
print(centers)
centers = trimesh.voxel.matrix_to_points(matrix, pitch=0.5, origin=(0.5/2,)*3)
print(centers)
centers = trimesh.voxel.matrix_to_points(matrix, pitch=0.5, origin=(0.5,)*3)
print(centers)
```

the output is

```
[[-0.25 -0.25 -0.25]
 [-0.25 -0.25  0.25]
 [-0.25  0.25 -0.25]
 [-0.25  0.25  0.25]
 [ 0.25 -0.25 -0.25]
 [ 0.25 -0.25  0.25]
 [ 0.25  0.25 -0.25]
 [ 0.25  0.25  0.25]]
[[0.  0.  0. ]
 [0.  0.  0.5]
 [0.  0.5 0. ]
 [0.  0.5 0.5]
 [0.5 0.  0. ]
 [0.5 0.  0.5]
 [0.5 0.5 0. ]
 [0.5 0.5 0.5]]
[[0.25 0.25 0.25]
 [0.25 0.25 0.75]
 [0.25 0.75 0.25]
 [0.25 0.75 0.75]
 [0.75 0.25 0.25]
 [0.75 0.25 0.75]
 [0.75 0.75 0.25]
 [0.75 0.75 0.75]]
```

This shows currently, the definition of origin is right top corner of the first element. (①in the figure)

<img src=https://user-images.githubusercontent.com/4310419/56053642-91693680-5d4c-11e9-9682-5d950a78d326.png width=30% />

Maybe this is too late (since -0.5 is added for a while ago), but for me ② or ③ looks more reasonable as origin. By flipping sign of +-0.5 or removing it.
What do you think about this?

https://github.com/mikedh/trimesh/blob/f785528788e89712a34074c552004a79adeb8264/trimesh/voxel.py#L657
https://github.com/mikedh/trimesh/blob/f785528788e89712a34074c552004a79adeb8264/trimesh/voxel.py#L629

**Comment by mikedh** [comment]
Not sure what the right answer here is. I think generally we probably want "points" to refer to the center of the voxel box?

**Comment by wkentaro** [comment]
I agree that design is the least confusing. In that case, we need to remove -0.5 and 0.5 and that is not backward compatible.

**Comment by mikedh** [comment]
Closed by #388
